tests/helpers/acceptance.js

```diff
@@ -8,6 +8,12 @@ var Promise           = require('../../lib/ext/promise');
 var root = path.resolve(__dirname, '..', '..');
 
 var PackageCache = require('../../tests/helpers/package-cache');
+var CommandGenerator = require('../../tests/helpers/command-generator');
+
+// We must link `ember-cli`.
+var npm = new CommandGenerator(require.resolve('npm/bin/npm-cli.js'));
+npm.invoke('link', { cwd: root });
```

Comment:
The original version of `PackageCache` handled the linking internally. However, the refactor here makes it 100% extractable which means consuming code should be responsible for managing its own links. That should remain the case, but `PackageCache` should enable consumer to be successful.

As it stands you're correct: because it relies on a single global link it doesn't enable you to test two builds simultaneously. I believed I had addressed all of those issues with moving the `package-cache.json` config but didn't think through the global `link` use case.

[For those following along, `cd ~/myproject && npm link ~/repos/ember-cli` actually creates a global link for `ember-cli` and then in `myproject` runs `npm link ember-cli`.](https://docs.npmjs.com/cli/link)

The only way to support multiple links is to do it _outside_ of the package manager itself, but that also gets thorny since `npm` possibly runs an `install` beforehand:
- [`npm` will `install` and build your package.](https://github.com/npm/npm/blob/latest/lib/link.js#L157-L165)
- [`yarn` just symlinks you.](https://github.com/yarnpkg/yarn/blob/0f1db96846badbbe3b43c470c05196eb29437fb4/src/cli/commands/link.js#L58)
on the code on `npm link`
- [`bower` just symlinks you.](https://github.com/bower/bower/blob/master/lib/commands/link.js)

Given that I can't (safely) blindly `link` you I elected to defer to the package manager's CLI. Thoughts?